### PR TITLE
[Doc] Add doc for the coordinated commits writer feature

### PR DESCRIPTION
**Pull request by dhruvarya-db**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [X] Other (Doc)

## Description

Documents the Coordinated Commits writer table feature as well as the DynamoDB Commit Coordinator implementation. Describes the process for enabling the feature for tables.

### CODE REVIEW

File: docs/source/delta-managed-commit.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+description: Learn about Delta Managed Commits.
+---
+
+# Delta Managed Commits
+
+.. warning:: This feature is available in preview in <Delta> 4.0. Since this feature is still in preview, it can undergo major changes. Furthermore, a table which has this preview feature enabled cannot be written to by future Delta releases until
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:7]: This is not rendering as expected in the github markdown viewer.

File: docs/source/delta-managed-commit.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+this feature is manually dropped from the table.
+
+The [Managed Commit](https://github.com/delta-io/delta/issues/2598) writer table feature defines a general protocol which can be extended to define custom commit protocols. The feature enables clients to designate "commit owners" for tables, which means that every commit to the table must be managed by that commit owner.
+The presence of a commit owner which is responsible for all commits to a table makes multi-cluster and multi-engine writes to tables safe. The commit owner for a table can be discovered by reading the table through the filesystem.
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:11]:
```suggestion
The presence of a commit owner which is responsible for all commits to a table makes multi-cluster and multi-engine writes to tables safe.
```

File: docs/source/delta-managed-commit.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+---
+description: Learn about Delta Managed Commits.
+---
+
+# Delta Managed Commits
+
+.. warning:: This feature is available in preview in <Delta> 4.0. Since this feature is still in preview, it can undergo major changes. Furthermore, a table which has this preview feature enabled cannot be written to by future Delta releases until
+this feature is manually dropped from the table.
+
+The [Managed Commit](https://github.com/delta-io/delta/issues/2598) writer table feature defines a general protocol which can be extended to define custom commit protocols. The feature enables clients to designate "commit owners" for tables, which means that every commit to the table must be managed by that commit owner.
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:10]:
```suggestion
[Managed Commit](https://github.com/delta-io/delta/issues/2598) is a new commit protocol which makes the commit process more flexible and pluggable by giving control of the commit to an external commit owner. Each managed commit table has a designated "commit owner" and all the commits to the table must go via it.

File: docs/source/delta-managed-commit.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+The presence of a commit owner which is responsible for all commits to a table makes multi-cluster and multi-engine writes to tables safe. The commit owner for a table can be discovered by reading the table through the filesystem.
+
+
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:13]: Should we add a compatibility section?

  ## Compatibility
  Managed Commit is a writer table feature.
  - That means only clients which understands this feature could write to a managed commit table. 
  - Older clients which do not understand this table feature could still read a managed commit table. Note that the read may give stale results depending on [table's commit owner backfill policy](#hyperLinkToBackfillSectionInProtocolRFC.md).
  ```

File: docs/source/delta-managed-commit.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+# DynamoDB Commit Owner
+
+<Delta> 4.0 also introduces a reference Commit Owner implementation which uses DynamoDB as the backend.
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:16]: Should this be 4.0-preview?

Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:16]:
```suggestion
<Delta> 4.0-preview also introduces a DynamoDB backed Commit Owner implementation.
```

File: docs/source/delta-managed-commit.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+## Quickstart Guide
+
+#. Create the DynamoDB table
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:20]:
```suggestion
# Create the DynamoDB table
```

Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:20]: What does `#.` do? Doesn't render in github UI.

File: docs/source/delta-managed-commit.md (hunk 1 above)
Comment (dhruvarya-db) [reply docs/source/delta-managed-commit.md:7]: It should render correctly in the docs: 
image

File: docs/source/delta-managed-commit.md (hunk 6 above)
Comment (dhruvarya-db) [reply docs/source/delta-managed-commit.md]: Changed to 1. to create an ordered list.

File: docs/source/delta-managed-commit.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+--conf spark.databricks.delta.managedCommits.commitOwner.ddb.awsCredentialsProviderName=<credentialsProviderName>
+```
+
+`<credentialsProviderName>` must be the fully qualified class name of the [AWS Credentials Provider](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSCredentialsProvider.html) (e.g. `com.amazonaws.auth.profile.ProfileCredentialsProvider`) which must be used for authenticating with the desired DynamoDB instance.
+
+### 3. Create a table with DynamoDB as the commit owner by running the following command:
+
+```sql
+CREATE TABLE <table_name> (id string) USING DELTA
+TBLPROPERTIES ('delta.managedCommits.commitOwner-preview' = 'dynamodb', 'delta.managedCommits.commitOwnerConf-preview' = '{\"managedCommitsTableName\": \"<dynamodb_table_name>\",\"dynamoDBEndpoint\": \"<dynamodb_region_endpoint>\"}');
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:55]: `managedCommitsTableName` should be `dynamoDBTableName` .

File: docs/source/delta-managed-commit.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+--packages io.delta:delta-spark_2.13:4.0.0,org.apache.hadoop:hadoop-aws:3.3.4 \
+--repositories https://oss.sonatype.org/content/repositories/iodelta-1147 \
+--conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension \
+--conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog \
+--conf spark.databricks.delta.managedCommits.commitOwner.ddb.awsCredentialsProviderName=<credentialsProviderName>
+```
+
+`<credentialsProviderName>` must be the fully qualified class name of the [AWS Credentials Provider](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSCredentialsProvider.html) (e.g. `com.amazonaws.auth.profile.ProfileCredentialsProvider`) which must be used for authenticating with the desired DynamoDB instance.
+
+### 3. Create a table with DynamoDB as the commit owner by running the following command:
```
Comment (prakharjain09) [inline docs/source/delta-managed-commit.md:51]: This is a title o a subsection.

```suggestion
### 3. Creating a managed-commit table with DynamoDB as the commit owner
```

Comment (dhruvarya-db) [reply docs/source/delta-managed-commit.md:51]: In that case:
"### 1. Create the DynamoDB table" --> "### 1. Creating the DynamoDB table"

"### 2. Launch a spark shell with the right packages:" --> "### 2. Launching the spark shell"
?

File: docs/source/versioning.md [hunk 9]
```diff
@@ -27,6 +27,7 @@ The following <Delta> features break forward compatibility. Features are enabled
    Domain metadata, [Delta Lake 3.0.0](https://github.com/delta-io/delta/releases/tag/v3.0.0),[Domain Metadata Spec](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#domain-metadata)
    Clustering, [Delta Lake 3.1.0](https://github.com/delta-io/delta/releases/tag/v3.1.0),[_](/delta-clustering.md)
    Type widening (Preview),[Delta Lake 3.2.0](https://github.com/delta-io/delta/releases/tag/v3.2.0),[_](/delta-type-widening.md)
+   Managed Commit (Preview),[Delta Lake 4.0.0](https://github.com/delta-io/delta/releases/tag/v4.0.0), [_](/delta-managed-commit.md)
```
Comment (prakharjain09) [inline docs/source/versioning.md:30]: Same as https://github.com/delta-io/delta/pull/3261#discussion_r1635492389 .

**Review by prakharjain09 — APPROVED** [review]

**Pull request merged**